blocks/README.md

```diff
@@ -154,6 +155,43 @@ Returns settings associated with a registered block.
 
 Returns settings associated with a registered control.
 
+## Backwards Compatibility
+
+When defining a block, you can choose to associate its compatibility with content shapes that had existed prior to the introduction of blocks. A block's `compatibility` property can include one or more objects which describe how the block can be initialized in cases where the editor cannot find another applicable handler.
+
+For example, consider a paragraph of text. A post's content may contain paragraphs, either explicitly with a paragraph `<p>` tag or inferred by newlines courtesy of the default [`wpautop` behavior](https://developer.wordpress.org/reference/functions/wpautop/). Prior to block parsing, content is normalized to apply `wpautop`, so thankfully we need only deal with the paragraph elements.
+
+We want to define a text block to enable controls specific to text (alignment, etc.), but we also want to apply this treatment to paragraph tags which were not saved with block demarcations.
+
+To do so, we'll declare compatibility with the paragraph tag when registering the block:
+
+```js
+registerBlock( 'core/text', {
+	attributes: {
+		content: html( 'p' )
+	},
+
+	compatibility: [
+		{
+			tagName: 'p'
```

Comment:
I agree, I think that not only should the compatibility matchers be as strict as possible, but also the parsing itself.  This should be the default option; we may also need a more flexible option like the `initialize` function proposed here, but I think we need to provide a standard way to handle the functions we expect.